### PR TITLE
Brackets service

### PR DESCRIPTION
**Pull request by uniphil**
Move business logic for buckets/brackets for any kind of metric into a service
Improve the colour service

### CODE REVIEW

File: app/scripts/services/brackets.coffee [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  getBracket: (val, metric) ->
+    unless typeof val in ['number', 'undefined']
+      throw new Error "val must be a number. Got: '#{val}' which is '#{typeof val}'"
+    if isNaN val  # NaN or undefined
+      'UNKNOWN'
+    else
+      switch metric
+
+        when 'AVG_MARK' then switch
```
Comment (esoliani) [inline app/scripts/services/brackets.coffee:20]: AVG_MARK should not be colorize. I will remove it next pull request.

File: app/scripts/services/brackets.coffee [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        else throw new Error "Unknown metric: '#{metric}'"
+
+
+  getMetric: (schoolType, criteria) ->
+    unless schoolType in ['primary', 'secondary']
+      throw new Error "Unknown school type '#{schoolType}'"
+    unless criteria in ['performance', 'improvement']
+      throw new Error "Unknown criteria '#{criteria}'"
+    switch schoolType
+      when 'primary' then switch criteria
```
Comment (esoliani) [inline app/scripts/services/brackets.coffee:59]: this is ok, only for sorting the dataset. not to paint the map

File: app/views/schoolFlyout.html [hunk 3]
```diff
@@ -16,30 +16,27 @@ <h3 translate="map.selected.ward">Ward</h3>
...
 </div>
-<div class="stat" ng-if="hovered.AVG_MARK">
-    <h3 translate="map.selected.avg-mark">Average Mark</h3>
-    <p class="number {{ hovered.AVG_MARK === null ?
-    'unknown' : hovered.AVG_MARK < 150 ?
-      'poor' : hovered.AVG_MARK < 200 ?
-        'medium' : 'good' }}">{{ hovered.AVG_MARK }}</p>
+<div class="stat" ng-if="hovered.AVG_MARK|exists">
+  <h3 translate="map.selected.avg-mark">Average Mark</h3>
+  <p class="number {{ getBracket(hovered.AVG_MARK, 'AVG_MARK') }}">
```
Comment (esoliani) [inline app/views/schoolFlyout.html:25]: this should not be colorize. I will remove it

File: app/scripts/services/brackets.coffee (hunk 1 above)
Comment (uniphil) [inline app/scripts/services/brackets.coffee:20]: oh good to know. It's currently used in `schoolFlyout` I think.

Comment (uniphil) [inline app/scripts/services/brackets.coffee:20]: oh, you already saw that. Nice :)

File: app/scripts/services/brackets.coffee (hunk 2 above)
Comment (uniphil) [inline app/scripts/services/brackets.coffee:59]: right, I forgot about that. The map is painted by pass rate, right?

Maybe we need separate `getSortMetric` and `getPaintMetric` functions?

**Pull request merged**